Sphinx 4.3 fails when importing PyQt5.QtCore.pyqtSignal

### Describe the bug

With Sphinx 4.3, usage of pyqtSignal will fail with the following error:
```
  File ".../lib/python3.8/site-packages/sphinx/util/docstrings.py", line 75, in prepare_docstring
    lines = s.expandtabs(tabsize).splitlines()
AttributeError: 'getset_descriptor' object has no attribute 'expandtabs'
```

Using exact same codebase, Sphinx 4.2 doesn't have such issue.

Investigating why, I stumbled upon the following:
```
> .../lib/python3.8/site-packages/sphinx/util/docstrings.py(75)prepare_docstring()
-> lines = s.expandtabs(tabsize).splitlines()
(Pdb) l
 70  	        ignore = 1
 71  	    else:
 72  	        warnings.warn("The 'ignore' argument to prepare_docstring() is deprecated.",
 73  	                      RemovedInSphinx50Warning, stacklevel=2)
 74  	
 75  ->	    lines = s.expandtabs(tabsize).splitlines()
 76  	    # Find minimum indentation of any non-blank lines after ignored lines.
 77  	    margin = sys.maxsize
 78  	    for line in lines[ignore:]:
 79  	        content = len(line.lstrip())
 80  	        if content:
(Pdb) interact
*interactive*
>>> s
<attribute '__doc__' of 'PyQt5.QtCore.pyqtSignal' objects>
>>> dir(s)
['__class__', '__delattr__', '__delete__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__get__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__name__', '__ne__', '__new__', '__objclass__', '__qualname__', '__reduce__', '__reduce_ex__', '__repr__', '__set__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__']
>>> 
now exiting InteractiveConsole...
(Pdb) 
make: *** [Makefile:19: html] Error 2
```

Removing references to `pyqtSignal` allows sphinx to successfully compile the documentation.

### How to Reproduce

Assuming you can quickstart the sphinxdocs folder yourself, insert the following snippet into a file and run sphinx against it.

```python
from PyQt5.QtCore import pyqtSignal

class test:
    """test"""
    x = pyqtSignal([tuple])
```

Don't forget to install PyQt5:
```
$ pip install PyQt5
```


### Expected behavior

_No response_

### Your project

Please follow reproduction steps.

### Screenshots

_No response_

### OS

Linux

### Python version

3.8

### Sphinx version

4.3.0

### Sphinx extensions

_No response_

### Extra tools

_No response_

### Additional context

_No response_

--- a/CHANGES
+++ b/CHANGES
@@ -18,6 +18,8 @@ Bugs fixed
 
 * #9838: autodoc: AttributeError is raised on building document for functions
   decorated by functools.lru_cache
+* #9879: autodoc: AttributeError is raised on building document for an object
+  having invalid __doc__ atribute
 
 Testing
 --------

--- a/sphinx/ext/autodoc/__init__.py
+++ b/sphinx/ext/autodoc/__init__.py
@@ -1704,7 +1704,7 @@ class ClassDocumenter(DocstringSignatureMixin, ModuleLevelDocumenter):  # type: 
         classdoc_from = self.options.get('class-doc-from', self.config.autoclass_content)
 
         docstrings = []
-        attrdocstring = self.get_attr(self.object, '__doc__', None)
+        attrdocstring = getdoc(self.object, self.get_attr)
         if attrdocstring:
             docstrings.append(attrdocstring)
 

--- a/sphinx/util/inspect.py
+++ b/sphinx/util/inspect.py
@@ -871,6 +871,13 @@ def getdoc(obj: Any, attrgetter: Callable = safe_getattr,
     * inherited docstring
     * inherited decorated methods
     """
+    def getdoc_internal(obj: Any, attrgetter: Callable = safe_getattr) -> Optional[str]:
+        doc = attrgetter(obj, '__doc__', None)
+        if isinstance(doc, str):
+            return doc
+        else:
+            return None
+
     if cls and name and isclassmethod(obj, cls, name):
         for basecls in getmro(cls):
             meth = basecls.__dict__.get(name)
@@ -879,7 +886,7 @@ def getdoc(obj: Any, attrgetter: Callable = safe_getattr,
                 if doc is not None or not allow_inherited:
                     return doc
 
-    doc = attrgetter(obj, '__doc__', None)
+    doc = getdoc_internal(obj)
     if ispartial(obj) and doc == obj.__class__.__doc__:
         return getdoc(obj.func)
     elif doc is None and allow_inherited:
@@ -888,7 +895,7 @@ def getdoc(obj: Any, attrgetter: Callable = safe_getattr,
             for basecls in getmro(cls):
                 meth = safe_getattr(basecls, name, None)
                 if meth is not None:
-                    doc = attrgetter(meth, '__doc__', None)
+                    doc = getdoc_internal(meth)
                     if doc is not None:
                         break
 

